payjoin/src/send/multiparty/mod.rs

```diff
@@ -22,18 +22,57 @@ mod error;
 pub struct SenderBuilder<'a>(v2::SenderBuilder<'a>);
 
 impl<'a> SenderBuilder<'a> {
-    pub fn new(psbt: Psbt, uri: PjUri<'a>) -> Self { Self(v2::SenderBuilder::new(psbt, uri)) }
-    pub fn build_recommended(self, min_fee_rate: FeeRate) -> Result<Sender, BuildSenderError> {
-        let v2 = v2::SenderBuilder::new(self.0 .0.psbt, self.0 .0.uri)
-            .build_recommended(min_fee_rate)?;
-        Ok(Sender(v2))
+    pub fn new(psbt: Psbt, uri: PjUri<'a>) -> Self {
+        Self(crate::send::v2::SenderBuilder::new(psbt, uri))
     }
+
+    pub fn build_recommended(self, min_fee_rate: FeeRate) -> Result<NewSender, BuildSenderError> {
+        let sender = self.0.build_recommended(min_fee_rate)?;
+        Ok(NewSender(sender))
+    }
+}
+
+pub struct NewSender(v2::NewSender);
+
+impl NewSender {
+    pub fn persist<P: Persister<Sender>>(
+        &self,
+        persister: &mut P,
+    ) -> Result<P::Token, ImplementationError> {
+        let sender =
+            Sender(v2::Sender { v1: self.0.v1.clone(), reply_key: self.0.reply_key.clone() });
+        persister.save(sender).map_err(ImplementationError::from)
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct SenderId(Url);
```

Comment:
Might makes more sense to call this `SenderToken` to be congruent with the other Token types

Reply:
One step ahead of you. Had this change locked and loaded

Reply:
Fixed